backend/oadoi.py

```diff
@@ -72,32 +65,32 @@ def create_oairecord(self, record):
                 print('no such paper for doi {doi}'.format(doi=doi))
                 return
 
-        url = record['url']
-
-        # just to speed things up a bit...
-        if paper.pdf_url == url:
-            return
-
-        identifier='oadoi:'+url
-        source = self.oadoi_source
-
-        if record['host_type'] == 'publisher':
-            url = doi_to_url(doi)
-            identifier = doi_to_crossref_identifier(doi)
-            source = self.crossref_source
-
-        record = BareOaiRecord(
-            paper=paper,
-            doi=doi,
-            pubtype=paper.doctype,
-            source=source,
-            identifier=identifier,
-            splash_url=url,
-            pdf_url=record['url'])
-        try:
-            paper.add_oairecord(record)
-            paper.update_availability()
-            # TODO re-enable this
-            #paper.update_index()
-        except (DataError, ValueError):
-            print('Record does not fit in the DB')
+        for oa_location in record.get('oa_locations') or []:
```

Comment:
`.get('oa_locations', [])` will do the same.

Reply:
It's not exactly equivalent: `{'oa_locations':None}.get('oa_locations', [])` returns `None`, which is not iterable, whereas  `{'oa_locations':None}.get('oa_locations') or []` returns `[]`, which is iterable. So I generally try to stick to the latter.

Reply:
Indeed, nice tip :)